build-logic/performance-testing/src/main/kotlin/gradlebuild/performance/PerformanceTestPlugin.kt

```diff
@@ -317,6 +380,55 @@ class PerformanceTestExtension(
         registerTestProject(testProject, T::class.java, configuration)
 
     fun <T : Task> registerTestProject(testProject: String, type: Class<T>, configurationAction: Action<in T>): TaskProvider<T> {
+        return doRegisterTestProject(testProject, type, configurationAction)
+    }
+
+    fun <T : Task> registerAndroidTestProject(testProject: String, type: Class<T>, configurationAction: Action<in T>): TaskProvider<T> {
+        return doRegisterTestProject(testProject, type, configurationAction) {
+            // AndroidStudio jvmArgs could be set per project, but at the moment that is not necessary
+            jvmArgumentProviders.add(getAndroidProjectJvmArguments(defaultAndroidStudioJvmArgs))
+            environment("JAVA_HOME", javaLauncher.get().metadata.installationPath.asFile.absolutePath)
+        }
+    }
+
+    private
+    fun getAndroidProjectJvmArguments(androidStudioJvmArgs: List<String>): CommandLineArgumentProvider {
+        val unpackAndroidStudio = project.tasks.named("unpackAndroidStudio", Copy::class.java)
+        val androidStudioInstallation = project.objects.newInstance<AndroidStudioInstallation>().apply {
+            studioInstallLocation.fileProvider(unpackAndroidStudio.map { it.destinationDir })
+        }
+        val autoDownloadAndroidStudio = isCiServer
+        val runInHeadlessMode = isCiServer
+        return object : AndroidStudioSystemProperties(androidStudioInstallation, autoDownloadAndroidStudio, project.providers) {
+            override fun asArguments(): Iterable<String> {
```

Comment:
Why is this method not part of `AndroidStudioSystemProperties`?

Reply:
I think it would be clearer to me if the whole logic would be in the class.

Reply:
Moved to the class